lib/spack/spack/solver/asp.py

```diff
@@ -3367,15 +3369,19 @@ def virtual_on_edge(self, parent_node, provider_node, virtual):
         dependencies[0].update_virtuals((virtual,))
 
     def reorder_flags(self):
-        """Order compiler flags on specs in predefined order.
-
-        We order flags so that any node's flags will take priority over
-        those of its dependents.  That is, the deepest node in the DAG's
-        flags will appear last on the compile line, in the order they
-        were specified.
+        """For each spec, determine the order of compiler flags applied to it.
 
         The solver determines which flags are on nodes; this routine
-        imposes order afterwards.
+        imposes order afterwards. The order is:
+
+        1. Compiler flags should always come first
+        2. Command-line flags should always come last
```

Comment:
this should probably be the last item in the list (i.e. this should be number 4).  I had to read the comment a few times and it was confusing to me that this was item 2.